### DISCUSSION TITLE
how to use plugin like 'babel-plugin-import'

### DISCUSSION THREAD

**Discussion by qianxuemin** [swc-project/swc#3042] [Q&A] [answered]
### Describe the feature

 how to use plugin like 'babel-plugin-import'

### Babel plugin or link to the feature description

https://www.npmjs.com/package/babel-plugin-import

**Comment by kwonoj** [comment]
This issue has moved into discussions, as this is a question which gh discussion is prepared for.

**Comment by Austaras** [comment, marked as answer]
The only possible solution is to chain a babel before swc, and I'm still wondering why such a legacy plugin has such a common name.

**Reply by kdy1** [reply]
It's my mistake :(

**Reply by Austaras** [reply]
What mistake?

**Reply by kwonoj** [reply]
> why such a legacy plugin has such a common name.

looks like this isn't core babel plugin even? I don't know if we'd like to support this in core swc.

**Reply by Austaras** [reply]
> > why such a legacy plugin has such a common name.
> 
> looks like this isn't core babel plugin even? I don't know if we'd like to support this in core swc.

No, it's a plugin mainly used by antd(a react component library built by Alibaba). It's built in maybe 2015 when tree shaking was still not quite a thing

**Reply by kwonoj** [reply]
I'd like to suggest to mark

> solution is to chain a babel before swc

as an answer or let users to come up with their own plugin approaches if possible. I honestly can't imagine core swc would implement this.

**Reply by magic-akari** [reply]
I'm not surprised it uses such a common name, since they also have a GitHub organization named [react-component](https://github.com/react-component).

**Reply by kdy1** [reply]
Ah, I was confused. I meant js plugin api of swc.

**Reply by hannadrehman** [reply]
We dont need really need this plugin if webpack treeshaking works as expected. but i have noticed our bundle sizes increases if i remove it. i am on a monorepo with hundreds of components and utilities this is sort of a must for us right now

**Reply by raulmarindev** [reply]
> The only possible solution is to chain a babel before swc, and I'm still wondering why such a legacy plugin has such a common name.

I don't know what "chain a babel before swc" means. Please, could someone point me in the right direction?

**Reply by moneko-o** [reply]
Maybe you can try this, although it is a JS plugin [swc-plugin-import-on-demand](https://www.npmjs.com/package/swc-plugin-import-on-demand)

**Comment by ankitchouhan1020** [comment]
Shameless plug. [swc-plugin-transform-import](https://github.com/ankitchouhan1020/swc-plugin-transform-import) does the similar thing. You can modify it according to your need. JS plugins are not the optimized for SWC but it does unblocks user to experiment with SWC.

**Reply by raulmarindev** [reply]
Has anyone been successful with this approach?

**Reply by ankitchouhan1020** [reply]
Hi @raulmarindev 

You might want to check this. https://github.com/swc-project/plugins/tree/main/packages/transform-imports

**Comment by Brooooooklyn** [comment]
https://github.com/speedy-js/speedy-native/blob/main/crates/speedy-transform/src/web_transform/babel_import.rs

Here's one implementation

**Comment by moneko-o** [comment]
Maybe you can try this, although it is a JS plugin [swc-plugin-import-on-demand](https://www.npmjs.com/package/swc-plugin-import-on-demand)
